cmd/zoekt-webserver/main.go

```diff
@@ -322,15 +331,30 @@ func addProxyHandler(mux *http.ServeMux, socket string) {
 	mux.Handle("/indexserver/", http.StripPrefix("/indexserver/", http.HandlerFunc(proxy.ServeHTTP)))
 }
 
-// shutdownOnSignal will listen for SIGINT or SIGTERM and call
-// srv.Shutdown. Note it doesn't call anything else for shutting down. Notably
-// our RPC framework doesn't allow us to drain connections, so it when
-// Shutdown is called all inflight RPC requests will be closed.
-func shutdownOnSignal(srv *http.Server) error {
+// shutdownSignalChan returns a channel which is listening for shutdown
+// signals from the operating system.
+func shutdownSignalChan() <-chan os.Signal {
 	c := make(chan os.Signal, 3)
```

Comment:
Just out of curiosity, why do we need capacity 3?

Reply:
we don't anymore, but the original version of this function could read upto 3 times on the channel. I will make it a parameter to make this more clear.